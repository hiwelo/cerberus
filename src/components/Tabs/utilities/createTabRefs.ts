import { useRef } from 'react';
import { TabsProps } from '../types';

/**
 * Creates a series of React Ref for each panel part of the Tabs defined for
 * this `Tabs` component
 *
 * @param tabs The dictionary of Tabs for the related Tabs component
 */
export function createTabRefs(tabs: TabsProps['tabs']): TabsProps['tabs'] {
  // Transforms the TabListDictionary as an array of tabs
  const tabsEntries = Object.entries(tabs);
  const tabsWithRefs: TabsProps['tabs'] = {};

  // Loops over each tab and creates a React Ref for each panel HTMLElement
  tabsEntries.forEach(([tabKey, tabInfo]) => {
    const refPanel = useRef<HTMLDivElement>(null);

    tabsWithRefs[tabKey] = { ...tabInfo, refPanel };
  });

  return tabsWithRefs;
}
