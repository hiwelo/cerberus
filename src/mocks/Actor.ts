import { Actor } from 'types/GitHubApi/Actor';

/**
 * Mocked Actor object
 */
export const mockedActor: Actor = {
  avatarUrl: 'https://i.pravatar.cc/300',
  login: 'test',
  url: 'https://github.com/test',
};
