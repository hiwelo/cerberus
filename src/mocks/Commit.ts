import { Commit } from '../types/GitHubApi/Commit';

/**
 * Mocked Commit object
 */
export const mockedCommit: Commit = {
  abbreviatedOid: '7638417',
  message: 'Test commit',
  messageBody: 'Test of message for the commit body',
  messageHeadline: 'Test of headline for the commit message',
  oid: '7638417db6d59f3c431d3e1f261cc637155684cd',
};
