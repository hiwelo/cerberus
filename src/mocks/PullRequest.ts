import {
  PullRequest,
  PullRequestConnection,
} from 'types/GitHubApi/PullRequest';
import { mockedActor, mockedCommit, mockedDeployment } from '.';

export const mockedPullRequest: PullRequest = {
  author: mockedActor,
  closed: false,
  locked: false,
  merged: false,
  number: 123,
  permalink: 'https://github.com/owner/repository/pull/123',
  title: 'Test of pull request',
  url: 'https://github.com/owner/repository/pull/123',
  commits: {
    edges: [
      {
        __typename: 'CommitEdge',
        node: {
          commit: {
            ...mockedCommit,
            deployments: {
              edges: [
                {
                  __typename: 'DeploymentEdge',
                  node: mockedDeployment,
                },
              ],
            },
          },
        },
      },
    ],
  },
};

export const mockedPullRequestConnection: PullRequestConnection = {
  edges: [
    {
      __typename: 'PullRequestEdge',
      node: mockedPullRequest,
    },
  ],
};
