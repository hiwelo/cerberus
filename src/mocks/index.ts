export { mockedActor } from './Actor';
export { mockedCommit } from './Commit';
export { mockedDeployment } from './Deployment';
export { mockedPullRequest } from './PullRequest';
