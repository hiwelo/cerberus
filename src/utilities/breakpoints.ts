/** List of breakpoints in use in our styled components */
export const breakpoints = {
  xs: '0px',
  md: '540px',
};
