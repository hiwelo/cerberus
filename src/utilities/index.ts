export { breakpoints } from './breakpoints';
export { listNodesFromConnection } from './listNodesFromConnection';
