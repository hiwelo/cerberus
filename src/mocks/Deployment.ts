import { Deployment, DeploymentConnection } from 'types/GitHubApi/Deployment';
import { mockedActor } from './Actor';
import { mockedCommit } from './Commit';

export const mockedDeployment: Deployment = {
  commit: mockedCommit,
  commitOid: '7638417db6d59f3c431d3e1f261cc637155684cd',
  creator: mockedActor,
  description: 'Test deployment',
  environment: 'test',
  latestStatus: {
    creator: mockedActor,
    description: 'Test environment',
    environmentUrl: 'https://example.com/',
    state: 'SUCCESS',
    updatedAt: '2019-12-20T20:11:45+00:00',
  },
  state: 'ACTIVE',
  task: 'deploy',
  updatedAt: '2019-12-20T20:11:45+00:00',
};

export const mockedDeploymentConnection: DeploymentConnection = {
  edges: [
    {
      __typename: 'DeploymentEdge',
      node: mockedDeployment,
    },
  ],
};
