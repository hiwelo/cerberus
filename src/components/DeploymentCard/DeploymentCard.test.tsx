import React from 'react';
import { testA11y, testSnapshot } from 'utilities/test';
import { mockedDeployment } from 'mocks';
import { DeploymentCard } from '.';

const TestContainer = <DeploymentCard item={mockedDeployment} />;

describe('DeploymentCard', () => {
  testA11y(TestContainer);
  testSnapshot(TestContainer);
});
