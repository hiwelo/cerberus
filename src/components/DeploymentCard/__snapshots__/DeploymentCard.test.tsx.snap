// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentCard should match snapshot 1`] = `
<span>
  deploy
</span>
`;
